scripts/apps/search/directives/SearchFilters.js

```diff
@@ -81,6 +81,23 @@ export const getDateFilters = (gettext) => [
         ],
         isEnabled: (searchConfig) => searchConfig.scheduled,
     },
+    {
+        labelBlock: gettext('Compliant lifetime'),
+        labelFrom: null,
+        labelTo: gettext('Need review before'),
+        fieldname: 'extra.compliantlifetime',
+        predefinedFilters: [
+            {
+                key: 'Next month',
+                label: gettext('Month'),
+            },
+            {
+                key: 'Next 3 months',
+                label: gettext('3 Months'),
+            }
+        ],
+        isEnabled: () => false,
+    }
```

Comment:
@tomaskikutis I added a filter here to use in getSearch, but it's not ideal as it's not used in superdesk-core, only in superdesk-fi. Any thoughts on this?

Reply:
`getDateFilters` is now responsible for 3 things: rendering available filters, displaying active filters at the top via tag service, and generating elasticsearch query from the URL. [image](https://user-images.githubusercontent.com/2076953/45466869-31ae9c80-b725-11e8-8b9f-36c929095445.png)

If I'm correct, you only need the last part: generating elasticsearch query. I suggest you refactor the search service(in client-core) and instead of having elasticsearch filters be generated from predefined list, you can implement life-cycle hooks for registering and de-registering functions which take a specific value in search field and add the elasticsearch filter to `facetrange` defined in search service. [here's a small example](https://user-images.githubusercontent.com/2076953/45467025-2d36b380-b726-11e8-9c7c-976947c36c9e.png)

`registerSearchTermHandlers` would be called both, by client-core search page and from superdesk-fi compliance review page.